lib/src/main/java/graphql/nadel/engine/blueprint/NadelFastSchemaTraverser.kt

```diff
@@ -38,242 +9,89 @@ import graphql.schema.GraphQLUnmodifiedType
  *
  * That's 4x faster.
  */
-internal class NadelFastSchemaTraverser {
+internal class NadelSchemaTraverser {
     fun traverse(
         schema: GraphQLSchema,
-        roots: List<String>,
-        visitor: Visitor,
+        roots: Iterable<String>,
+        visitor: NadelSchemaTraverserVisitor,
     ) {
-        val queue = roots
+        val queue: MutableList<NadelSchemaTraverserElement> = roots
```

Comment:
We aren't making any more objects than before.

The old type was `List<Pair<GraphQLNamedSchemaElement, GraphQLNamedType>>` i.e. we are replacing `Pair` with our own `NadelSchemaTraverserElement`